src/Workspaces/CSharp/Portable/CodeGeneration/ParameterGenerator.cs

```diff
@@ -92,23 +92,21 @@ internal static ParameterSyntax GetParameter(IParameterSymbol p, CodeGenerationO
         private static SyntaxTokenList GenerateModifiers(
             IParameterSymbol parameter, bool isFirstParam)
         {
+            SyntaxTokenList list = CSharpSyntaxGenerator.GetParameterModifiers(parameter.RefKind);
+
             if (isFirstParam &&
-                parameter.ContainingSymbol is IMethodSymbol &&
-                ((IMethodSymbol)parameter.ContainingSymbol).IsExtensionMethod)
+                parameter.ContainingSymbol is IMethodSymbol methodSymbol &&
+                methodSymbol.IsExtensionMethod)
             {
-                return SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.ThisKeyword));
+                list = list.Add(SyntaxFactory.Token(SyntaxKind.ThisKeyword));
             }
 
             if (parameter.IsParams)
             {
-                return SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.ParamsKeyword));
+                list = list.Add(SyntaxFactory.Token(SyntaxKind.ParamsKeyword));
             }
 
-            return parameter.RefKind == RefKind.None
-                ? new SyntaxTokenList()
-                : parameter.RefKind == RefKind.Out
-                    ? SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.OutKeyword))
-                    : SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.RefKeyword));
+            return list;
```

Comment:
where does out/ref get added now?

Reply:
in CSharpSyntaxGenerator.GetParameterModifiers(RefKind)

---
In reply to: [141181107](https://github.com/dotnet/roslyn/pull/22285#discussion_r141181107) [](ancestors = 141181107)